### ISSUE TITLE
Errors with MySQL Setup When Running `yarn docker:build` on Ubuntu 16.04 Instance

### ISSUE THREAD

**Issue by jonathanhudak** [postlight/headless-wp-starter#115]
Hi! This project seems awesome! I got it up and running locally on MacOS fine using Docker, but I've run into some issues setting it up on an Ubuntu instance using AWS Lightsail.

Here's what I've done so far:

Installed Docker, Node 7 (as per the .nvmrc file), and yarn
https://gist.github.com/hudakdidit/10c4f80ad0a5236f4f03f84c3930c8d6

I then needed to run`sudo usermod -a -G root $USER` to set permissions to run Docker.

Cloned this repo into the root folder of the instance. 

From cloned headless-wp-starter/ directory I ran `yarn docker:build` and here are some highlights of the output:

It seems MySQL 5.7 is first installed, then during step 19 it is uninstalled and upgraded:
```bash
Step 19/21 : RUN apt-get -qy install mysql-server
 ---> Running in f194ad9d3f88
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libmecab2 mecab-ipadic mecab-ipadic-utf8 mecab-utils mysql-client
  mysql-common mysql-community-client mysql-community-client-core
  mysql-community-server mysql-community-server-core
The following packages will be REMOVED:
  mysql-client-5.7 mysql-client-core-5.7
The following NEW packages will be installed:
  libmecab2 mecab-ipadic mecab-ipadic-utf8 mecab-utils mysql-community-client
  mysql-community-client-core mysql-community-server
  mysql-community-server-core mysql-server
The following packages will be upgraded:
  mysql-client mysql-common
2 upgraded, 9 newly installed, 2 to remove and 5 not upgraded.
```

I then see that mysql 8.0 attempt to run and fails.
```bash
/usr/share/mysql-8.0/mysql-helpers: line 73:   280 Killed                  mysqld --defaults-group-suffix="$2" --user=mysql --init-file="$1" --socket="$tmpdir/mysqld.sock" --pid-file="$tmpdir/mysqld.pid" > /dev/null 2>&1
dpkg: error processing package mysql-community-server (--configure):
 installed mysql-community-server package post-installation script subprocess returned error exit status 137
dpkg: dependency problems prevent configuration of mysql-server:
 mysql-server depends on mysql-community-server (= 8.0.11-1ubuntu18.04); however:
  Package mysql-community-server is not configured yet.

dpkg: error processing package mysql-server (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 mysql-community-server
 mysql-server
E: Sub-process /usr/bin/dpkg returned an error code (1)
The command '/bin/sh -c apt-get -qy install mysql-server' returned a non-zero code: 100
```
I updated the Dockerfile [currently on line 27](https://github.com/postlight/headless-wp-starter/blob/master/Dockerfile#L27) to `RUN apt-get -qy install mysql-server-5.7`

I ran `docker system prune -a` to start clean again (I think??)

Re-ran `yarn docker:build` and now I do not see the same error at the end of the build and it seems to build successfully. I *do* see the following:
```bash
Setting up mysql-server-5.7 (5.7.22-0ubuntu18.04.1) ...
invoke-rc.d: could not determine current runlevel
invoke-rc.d: policy-rc.d denied execution of stop.
update-alternatives: using /etc/mysql/mysql.cnf to provide /etc/mysql/my.cnf (my.cnf) in auto mode
Renaming removed key_buffer and myisam-recover options (if present)
Cannot stat file /proc/6/fd/0: Permission denied
Cannot stat file /proc/6/fd/1: Permission denied
Cannot stat file /proc/6/fd/2: Permission denied
Cannot stat file /proc/6/fd/3: Permission denied
Cannot stat file /proc/6/fd/4: Permission denied
Cannot stat file /proc/6/fd/5: Permission denied
Cannot stat file /proc/6/fd/6: Permission denied
Cannot stat file /proc/6/fd/7: Permission denied
Cannot stat file /proc/6/fd/8: Permission denied
Cannot stat file /proc/6/fd/9: Permission denied
Cannot stat file /proc/6/fd/10: Permission denied
Cannot stat file /proc/6/fd/11: Permission denied
Cannot stat file /proc/6/fd/12: Permission denied
Cannot stat file /proc/6/fd/13: Permission denied
Cannot stat file /proc/6/fd/14: Permission denied
Cannot stat file /proc/6/fd/15: Permission denied
Cannot stat file /proc/6/fd/16: Permission denied
Cannot stat file /proc/6/fd/17: Permission denied
Cannot stat file /proc/6/fd/18: Permission denied
Cannot stat file /proc/6/fd/19: Permission denied
Cannot stat file /proc/6/fd/20: Permission denied
Cannot stat file /proc/6/fd/21: Permission denied
Cannot stat file /proc/6/fd/22: Permission denied
Cannot stat file /proc/6/fd/23: Permission denied
Cannot stat file /proc/6/fd/24: Permission denied
Cannot stat file /proc/6/fd/25: Permission denied
Cannot stat file /proc/6/fd/26: Permission denied
Initialization of mysqld failed: 0
Warning: Unable to start the server.
invoke-rc.d: could not determine current runlevel
invoke-rc.d: policy-rc.d denied execution of start.
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Removing intermediate container 68a321e5fd8b
 ---> 315b34692bf0
Step 20/21 : EXPOSE 8080
 ---> Running in ca0ee5767b30
Removing intermediate container ca0ee5767b30
 ---> 1d459f0c7575
Step 21/21 : CMD [ "./start_docker.sh" ]
 ---> Running in 4cd3975f07ef
Removing intermediate container 4cd3975f07ef
 ---> 86072b51eff1
Successfully built 86072b51eff1
Successfully tagged wp_headless:latest
```

Then I run `yarn docker:start` and see `ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)` so the database setup fails.

```bash
yarn run v1.7.0
warning You are using Node "7.10.1" which is not supported and may encounter bugs or unexpected behavior. Yarn supports the following semver range: "^4.8.0 || ^5.7.0 || ^6.2.2 || >=8.0.0"
$ docker run -v $(pwd)/data:/var/lib/mysql -v $(pwd):/usr/src/app -p 8080:8080 -it wp_headless
 * Starting MySQL database server mysqld                                                                                                                                                                                                      No directory, logging in with HOME=/
                                                                                                                                                                                                                                       [fail]

 This will replace your current WordPress install. Are you sure you want to do this? (yes/no) [no]:
 > y

?  Do you have an existing database you'd like to use and configure yourself? (y/n):  n
 [Exec] Running mysql -uroot -proot -h localhost -e 'create user if not exists wp_headless@localhost identified with mysql_native_password by "wp_headless"'
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
```

Anyone have any ideas on how I can get it working? Seems to be a permissions issue, but I am not quite sure.

Thanks so much!

**Comment by khovansky-al** [comment]
@hudakdidit unlike you I wasn't able to even get it running locally in Docker on MacOS due to MySQL misconfiguration
Your issue can likely be solved by adding this to the Dockerfile before the MySQL section
```
# Avoid ERROR: invoke-rc.d: policy-rc.d denied execution of start.
RUN sed -i "s/^exit 101$/exit 0/" /usr/sbin/policy-rc.d
```
But even then - this won't work since mysql is not starting properly with this.
MySQL 8 seems to work fine but the root password is obviously not set. I ended up editing the RoboFile to use empty password for the installation while I'm tinkering with this repo.
If this project is production-worthy, it might be better to deploy DB in a separate container and link backend with DB using docker-compose instead of keeping everything together.

**Comment by modelm** [comment]
I don't think the .nvmrc file is valid any more. I am running this on OS X 10.13.6 with Node v10.9.0 - so I would try the latest node on Ubuntu as well, but that may be unrelated to the MySQL errors (which I can't speak to quite yet - I'll update if/when I get a chance to try and reproduce this on Ubuntu 16.04).

**Comment by alexiakl** [comment]
This should be solved in the following merge: #160

**Comment by jonathanhudak** [comment]
Nice! I'll give it a try again when I have the chance.
